src/Controls/src/Core/Border/Border.cs

```diff
@@ -305,7 +305,10 @@ protected override void OnPropertyChanged([CallerMemberName] string? propertyNam
 			if (propertyName == HeightProperty.PropertyName ||
 				propertyName == WidthProperty.PropertyName ||
 				propertyName == StrokeShapeProperty.PropertyName)
+			{
 				Handler?.UpdateValue(nameof(IBorderStroke.Shape));
+				UpdateStrokeShape();
+			}
```

Comment:
I also see the `UpdateStrokeShape` method has a condition for:

```cs
if (StrokeShape is Shape strokeShape && StrokeThickness == 0)
{
    strokeShape.StrokeThickness = StrokeThickness;
}
```

Do we know why we only apply the thickness if it is 0? What happens if the value is say 10?

Reply:
At the very least, that `0` scares me so a comment is probably worth adding since future someone (aka this Matthew guy) will ask this and maybe do terrible things.

I also wonder if this is worth adding a basic/normal unit test for this? Then there can be also caught there before running all the UI tests: 

 - Core: https://github.com/dotnet/maui/blob/main/src/Core/tests/UnitTests/Views/BorderTests.cs
 - Controls: https://github.com/dotnet/maui/blob/main/src/Controls/tests/Core.UnitTests/BorderUnitTests.cs

I see there like 10 tests total which is surprisingly (I know why so this is a fake surprise) low for this control that is supposedly the future of all containers and borders and other amazingness.

I think a few tests to make sure the thicknesses are propagated correctly to the shape and maybe any other properties that seem to all be intermingled with the shape property.